test/cc_shared_library/BUILD.bazel

```diff
@@ -0,0 +1,30 @@
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+load("@rules_rust//rust:defs.bzl", "rust_static_library")
+
+rust_static_library(
+    name = "rust_lib",
+    srcs = ["lib.rs"],
+    edition = "2021",
+)
+
+cc_library(
+    name = "c_lib",
+    srcs = ["lib.c"],
+    hdrs = ["lib.h"],
+    deps = [":rust_lib"],
+)
+
+# Tests that cc_shared_library correctly traverses into
+# `rust_static_library` when linking.
+cc_shared_library(
```

Comment:
Will this lead to the same issue as https://github.com/bazelbuild/rules_rust/issues/2359 if `experimental_cc_shared_library` is not set?

Reply:
Do you mean if people are querying it as an external repository, where the root workspace does not have this set?

Reply:
yeah

Reply:
I suppose it will, although in 7.0.0 this is enabled by default so it shouldn't be an issue with the latest version.

Reply:
@illicitonion given that the issue was referring to a cquery within the `rules_rust` repo, do you think this is an acceptable change? I do like the `procides` changes and love the test. I just don't wanna immediately regress something here.

Reply:
I sent out https://github.com/bazelbuild/rules_rust/pull/2398 which add tests for what I think we should be aiming to support (i.e. `bazel query ...` and `bazel cquery ...` from inside the `rules_rust` repo) - if we have more things we want to support (e.g. `bazel query @rules_rust//...` from a repo that depends on `rules_rust`) , we should add tests to CI for them.

This PR passes the tests I sent out, so I'm happy :)